src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,128 @@
+import wolox.training.exceptions.EmptyNeccessaryAttributesException;
+import wolox.training.exceptions.UserAlreadyExistsException;
+import wolox.training.exceptions.UserDoesNotExistException;
+import wolox.training.exceptions.UserDoesNotHaveAnyBookException;
+import wolox.training.exceptions.UserDoesNotHaveTheBookException;
+import wolox.training.exceptions.UserIdMismatchException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("api/users")
+public class UserController {
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    @GetMapping
+    public Iterable findAll() {
+        return userRepository.findAll();
+    }
+
+    @GetMapping("/username/{username}")
+    public User findByUsername(@PathVariable final String username) {
+        return userRepository.findOneByUsername(username)
+            .orElseThrow(() -> new UserDoesNotExistException(username));
+    }
+
+    @GetMapping("/{id}")
+    public User findOne(@PathVariable final Long id) {
+        return userRepository.findById(id).orElseThrow(() -> new UserDoesNotExistException(id));
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody final User user) {
+        if ((user.getName().isEmpty()) || (user.getUsername().isEmpty()) || (user.getBirthDate() == null)) {
+            throw new EmptyNeccessaryAttributesException();
+        }
+        if (userRepository.findOneByUsername(user.getUsername()).isPresent()) {
+            throw new UserAlreadyExistsException(user.getUsername());
+        }
+        return userRepository.save(user);
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteById(@PathVariable final Long id) {
+        userRepository.findById(id)
+            .orElseThrow(() -> new UserDoesNotExistException(id));
+        userRepository.deleteById(id);
+    }
+
+    @DeleteMapping("/username/{username}")
+    public void deleteByUsername(@PathVariable final String username) {
+        User userToDelete = userRepository.findByUsername(username)
+            .orElseThrow(() -> new UserDoesNotExistException(username));
+        userRepository.deleteById(userToDelete.getId());
+    }
+
+    @PutMapping("/{id}")
+    public User updateUserAttributes(@RequestBody final User user, @PathVariable final Long id) {
+        if (user.getId() != id) {
+            throw new UserIdMismatchException(user.getId(), id);
+        }
+        User userToUpdate = userRepository.findById(user.getId())
+            .orElseThrow(() -> new UserDoesNotExistException(user.getUsername()));
+        userToUpdate.setBirthDate(user.getBirthDate());
+        userToUpdate.setName(user.getName());
+        userToUpdate.setUsername(user.getUsername());
+        return userRepository.save(userToUpdate);
```

Comment:
Why you set these fields?
You can update the user using save
`return userRepository.save(user)`